### PR TITLE
Update dirtyfrag workaround to remove mitigation on fixed OCP versions

### PR DESCRIPTION
**Pull request by tsatam**
### Which issue this PR addresses:

Fixes [ARO-27292](https://redhat.atlassian.net/browse/ARO-27292)

### What this PR does / why we need it:

Updates the dirtyfrag workaround implementation to remove the mitigation applied to cluster control plane nodes on fixed OCP versions. 

See #4820 for a previous implementation of this pattern for the copyfail mitigations.
See https://access.redhat.com/solutions/7142250 for the specific fixed OCP versions chosen here. 

### Test plan for issue:

- [x] Unit tests were added for this and it and all existing tests continue to pass
- [x] All E2E tests continue to pass

### Is there any documentation that needs to be updated for this PR?

No

### How do you know this will function as expected in production? 

Above testing and validation

### CODE REVIEW

File: pkg/operator/controllers/workaround/dirtyfragworkaround.go [hunk 1]
```diff
@@ -76,6 +86,15 @@ func (a *dirtyfragworkaround) IsRequired(ctx context.Context, clusterVersion ver
 		}
 	}
 
+	if clusterVersion.Gt(version.NewVersion(4, 22, 0)) {
```
[the remark this answers is not in the document]
Comment (tsatam) [reply pkg/operator/controllers/workaround/dirtyfragworkaround.go]: addressed in latest commit

**Review by cloudygreybeard — APPROVED** [review]
Clean follow-up to the copyfail version-gating in #4820— same pattern applied to dirtyfrag, and the `Gt` removal is a nice resolution of the semantics question I raised there. Replacing it with explicit `!Lt` at the call sites is simpler than renaming or adding comments. Cheers for tidying that up.

**Version gating is correct.** The `dirtyfragFixedPatchVersions` map combined with the `>= 4.22` guard covers the full version space. Clusters below the fix threshold keep the mitigation; clusters at or above get it removed automatically.

**Mid-upgrade behaviour remains safe.** Same as copyfail— `GetClusterVersion()` returns the last `CompletedUpdate`, so the mitigation stays in place until the upgrade finishes.

**Test coverage is thorough.** Boundary tests for each listed minor at the exact fix version and one below, plus the 4.22 cutoff and the unlisted-minor (4.11) fall-through case.

LGTM.

**Review by tuxerrante — APPROVED** [review]
LGTM — clean follow-up to the copyfail version-gating in #4820.

**Version boundaries verified** against [ARO-27292](https://redhat.atlassian.net/browse/ARO-27292) / Red Hat solution article 7142250 — all fixed z-versions match, and EOL streams without fixes (4.17, 4.15, 4.13) are correctly excluded from the map so all their z-versions continue to receive the mitigation.

**`Gt` removal is a good call** — the old implementation (`!Lt`) was semantically `Gte`, not `Gt`. Replacing with explicit `!Lt()` avoids the naming confusion. Verified no other callers exist in the codebase.

**Minor suggestion (not blocking):** the table-driven subtests in both `dirtyfragworkaround_test.go` and `copyfailworkaround_test.go` don't call `t.Parallel()`. Each subtest creates its own client and workaround instance with no shared mutable state, so they're safe to parallelize. Adding `t.Parallel()` at both the top-level test and inside each `t.Run` subtest would speed up the test suite as the number of version-boundary cases grows.
